pkg/psalabelsyncer/podsecurity_label_sync_controller.go

```diff
@@ -250,19 +251,22 @@ func (c *PodSecurityAdmissionLabelSynchronizationController) syncNamespace(ctx c
 		return fmt.Errorf("unknown PSa level for namespace %q", ns.Name)
 	}
 
-	nsCopy := ns.DeepCopy()
+	nsApplyConfig, err := corev1apply.ExtractNamespace(ns, controllerName)
+	if err != nil {
+		return fmt.Errorf("failed to convert namespace to its apply configuration form: %w", err)
+	}
 
 	var changed bool
 
 	if c.shouldEnforce {
```

Comment:
I think it would be easier to debug by checking in metadata.managedFields

```yaml
  managedFields:
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:labels:
          f:hypershift.openshift.io/monitoring: {}
          f:openshift.io/cluster-monitoring: {}
      f:spec: {}
    manager: hypershift
    operation: Apply
    time: "2023-07-26T18:09:23Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:openshift.io/sa.scc.mcs: {}
          f:openshift.io/sa.scc.supplemental-groups: {}
          f:openshift.io/sa.scc.uid-range: {}
        f:labels:
          f:pod-security.kubernetes.io/audit: {}
          f:pod-security.kubernetes.io/audit-version: {}
          f:pod-security.kubernetes.io/warn: {}
          f:pod-security.kubernetes.io/warn-version: {}
    manager: cluster-policy-controller
    operation: Update
    time: "2023-07-26T18:09:23Z"
```

Notice how labels are subdivided by individual name (two owners in that snippet), so you can see if *your* controller set the value.

Reply:
Isn't it better to work with the library rather than our own unstructured handling?